src/core/lib/promise/activity.h

```diff
@@ -199,7 +199,16 @@ class Activity : public Orphanable {
   //   locked
   // - back up that assertation with a runtime check in debug builds (it's
   //   prohibitively expensive in non-debug builds)
-  static Activity* current() { return g_current_activity_; }
+  static Activity*& current() {
```

Comment:
This should not be returning a reference from a public method.
Doing so invites mutating private state from anywhere in the codebase.

Reply:
Uses a private refToCurrent function for the mutation